src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -686,6 +686,9 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
     // TODO: Handle HostComponent tags here as well and call pushHostContext()?
```

Comment:
What about this TODO btw?

Reply:
Looks like the same issue, but for HostComponents. This whole function is a code smell, IMO. Ideally we'd bail out on low priority before we ever call `performUnitOfWork`. I tried this out on my big refactor PR from earlier this summer. Will revisit when we add back resuming.